smithy-aws-protocol-tests/model/restXml/document-structs.smithy

```diff
@@ -143,12 +143,17 @@ apply SimpleScalarProperties @httpResponseTests([
     },
     {
         id: "SimpleScalarPropertiesWithEscapedCharacter",
-        documentation: "Serializes string with escaping",
+        documentation: """
+        Serializes string with escaping.
+
+        This validates the three escape types: literal, decimal and hexadecimal. It also validates that unescaping properly
+        handles the case where unescaping an & produces a newly formed escape sequence (this should not be re-unescaped).
+        """,
         protocol: restXml,
         code: 200,
         body: """
               <SimpleScalarPropertiesInputOutput>
-                  <stringValue>&lt;string&gt;</stringValue>
+                  <stringValue>escaped data: &amp;lt;&#xD;&#10;</stringValue>
```

Comment:
I tested existing implementations and most will actually return the newline character un-escaped, though it's still reasonable to have this as a response test. Still, you should mark this as applying to clients only, and ideally add a server variant that reflects the existing behavior.

I also tested S3 by creating an object with this as a key, and it seems they prefer to url encode: `escaped+data%3A+%26lt%3B%0D%0A`

Reply:
yeah good call, this should be clients only. Re: S3, are they returning that in an XML payload?

Reply:
Yeah. Create an object with your test string as a key and then list objects